internal/app/machined/pkg/controllers/k8s/kubelet_static_pod.go

```diff
@@ -189,12 +187,18 @@ func (ctrl *KubeletStaticPodController) refreshPodStatus(ctx context.Context, r
 	podsSeen := map[string]struct{}{}
 
 	for _, pod := range podList.Items {
-		if pod.Metadata.Annotations.ConfigSource != "file" {
+		pod := pod
+
+		switch pod.Metadata.Annotations.ConfigSource {
```

Comment:
this is the actual fix, everything else is less important

kubelet package doesn't have these constants exported unfortunately properly, so we have to hardcode them here